src/main/java/com/mkul/game/ox/Players.java

```diff
@@ -0,0 +1,73 @@
+package com.mkul.game.ox;
+
+import java.util.LinkedList;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ThreadLocalRandom;
+
+class Players {
```

Comment:
I think you should create builder for Players not for a Player inside Players. Players should only exists when there are 2 of them. You can create players outside Players class and then pass them to Players builder. Also I think you can delete players list and change it to field for player1 and field for player2 (or something else, I'm not sure cause I don't implement it in my project yet).